### PR TITLE
chore: Merge from main

### PR DESCRIPTION
**Pull request by palpatim**
#### Description of changes

Merge `main` into `feature/rds-support-preview2`.

**NOTE**: After the 2024-10-12 release the `feature/rds-support-preview2` branch failed to build, so I re-merged from main. The merge conflicts are actually easy to resolve, but there’s an interface incompatibility in the feature branch: 9297fa5cd changed the `ConfiguredAuthProviders` interface to replace `identityPoolId` with `hasIdentityPoolId`. It seems to be a relatively straightforward fix (convert call sites to use the boolean flag; update `convertAuthRoleToVtl` to pull `identityPoolId` from `ctx.stash`, but  please review to make sure it looks right. (Notably, I did not find where the RDS resolver chain sets `identityPoolId` in the stash. I’m assuming it does, but please confirm.)

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-graphql-auth-transformer/src/vtl-generator/rds/resolvers/common.ts [hunk 1]
```diff
@@ -39,18 +39,23 @@ export const generateAuthRulesFromRoles = (
...
   });
   return expressions;
 };
 
-const convertAuthRoleToVtl = (role: RoleDefinition, fields: string[], hideAllowedFields = false, identityPoolId?: string): Expression => {
+const convertAuthRoleToVtl = (
+  role: RoleDefinition,
+  fields: string[],
+  hideAllowedFields = false,
+  hasIdentityPoolId: boolean
```
Comment (sundersc) [inline packages/amplify-graphql-auth-transformer/src/vtl-generator/rds/resolvers/common.ts:57]: Shouldn't the last argument be optional or have a default value?

**Review by palpatim — COMMENTED** [review]
Marker review

File: packages/amplify-graphql-auth-transformer/src/vtl-generator/rds/resolvers/common.ts (hunk 1 above)
Comment (phani-srikar) [reply packages/amplify-graphql-auth-transformer/src/vtl-generator/rds/resolvers/common.ts]: Better to keep the required args before ones with default. Keeping `hasIdentityPoolId` as required makes sense after inspecting the call sites.

Comment (palpatim) [reply packages/amplify-graphql-auth-transformer/src/vtl-generator/rds/resolvers/common.ts]: Sounds good, will re-order the args

**Review by phani-srikar — APPROVED** [review]

**Review by sundersc — APPROVED** [review]

**Pull request merged**